source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -869,6 +873,61 @@ ServerContextImpl::ServerContextImpl(Stats::Scope& scope,
   }
 }
 
+void ServerContextImpl::populateServerNamesMap(TlsContext& ctx, int pkey_id) {
+  if (ctx.cert_chain_ == nullptr) {
+    return;
+  }
+
+  auto populate = [&](const std::string& sn) {
+    std::string sn_pattern = sn;
+    if (absl::StartsWith(sn, "*.")) {
+      sn_pattern = sn.substr(1);
+    }
+    PkeyTypesMap pkey_types_map;
+    // Multiple certs with different key type are allowed for one server name pattern.
```

Comment:
@ggreenway I found two different cert might have the same SAN entry, how do we handle this case? In current implementation ,it fails to load certs and envoy will crash.

The example in real world:
https://www.gstatic.com/: Its SANs contains *.gstatic.com
https://aa.google.com/: it has "*.gstatic.com" as well.

Reply:
@ggreenway kindly ping

Reply:
Envoy will crash? Or just fail to load the certs? Obviously it shouldn't crash, so if it does that needs to be handled properly and turned into an error.

I'd prefer to disallow duplicates, because if there are duplicates then we need a policy on which cert will be chosen, and the code will be more complicated.

Reply:
@ggreenway  "Crash" might not be a proper word. I mean an exception will be raised, and envoy fails to run. https://github.com/envoyproxy/envoy/blob/3edf9207527a90e926195e3a788a786c4ffa1e2d/source/extensions/transport_sockets/tls/context_impl.cc#L890-L895

thanks, it's indeed not easy to find a policy on choosing cert, I agree with that we disallow duplicates.